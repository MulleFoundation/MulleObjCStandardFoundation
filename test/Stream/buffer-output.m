#import <MulleObjCStandardFoundation/MulleObjCStandardFoundation.h>


@interface SimpleStream : NSObject <MulleObjCOutputStream>
@end


@implementation SimpleStream

- (void) mulleWriteBytes:(void *) bytes
                  length:(NSUInteger) length
{
   printf( "%.*s\n", (int) length, bytes);
}

@end


int   main( void)
{
   MulleObjCBufferedOutputStream   *stream;
   SimpleStream                    *simple;
   int                             i;

   simple = [SimpleStream object];
   stream = [[[MulleObjCBufferedOutputStream alloc] initWithOutputStream:simple
                                                             flushLength:32] autorelease];
   for( i = 0; i < 16; i++)
      [stream mulleWriteBytes:"VfL Bochum 1848"
                       length:-1];
   [stream flush];

   return( 0);
}
