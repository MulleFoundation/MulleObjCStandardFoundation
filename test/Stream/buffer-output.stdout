VfL Bochum 1848VfL Bochum 1848VfL Bochum
 1848VfL Bochum 1848VfL Bochum 1848VfL B
ochum 1848VfL Bochum 1848VfL Bochum 1848
VfL Bochum 1848VfL Bochum 1848VfL Bochum
 1848VfL Bochum 1848VfL Bochum 1848VfL B
ochum 1848VfL Bochum 1848VfL Bochum 1848

